### PR TITLE
remove DomainStatus handlers from zedagent to avoid race condition

### PR DESCRIPTION
**Pull request by eriknordmark**
Signed-off-by: eriknordmark <erik@zededa.com>

### CODE REVIEW

File: pkg/pillar/cmd/zedagent/handlemetrics.go [hunk 1]
```diff
@@ -120,163 +120,21 @@ func ExecuteXlInfoCmd() map[string]string {
 	return dict
 }
 
-// Shadow copy of suscription to determine info for deletes. Key is UUID
-var domainStatus map[string]types.DomainStatus
+// Helper function
+func getAppInterfaceList(aiStatus types.AppInstanceStatus) []string {
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/zedagent/handlemetrics.go]: This looks like it should be a method of struct AppInstanceStatus ?? 
     aiStatus.getAppInterfaceList()

File: pkg/pillar/cmd/zedagent/handlemetrics.go [hunk 2]
```diff
@@ -716,7 +574,7 @@ func PublishMetricsToZedCloud(ctx *zedagentContext, cpuMemoryStat [][]string,
 		availableMemoryPercent := 100.0 - usedMemoryPercent
 		ReportAppMetric.Memory.AvailPercentage = availableMemoryPercent
 
-		appInterfaceList := ReadAppInterfaceList(aiStatus.DomainName)
+		appInterfaceList := getAppInterfaceList(aiStatus)
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/zedagent/handlemetrics.go]: as mentioned above:
    appInterfaceList := aiStatus.getAppInterfaceList()

File: pkg/pillar/cmd/zedagent/handlemetrics.go [hunk 3]
```diff
@@ -1569,7 +1408,7 @@ func PublishAppInfoToZedCloud(ctx *zedagentContext, uuid string,
 		// Mostly reporting the UP status
 		// We extract the appIP from the dnsmasq assignment
 		interfaces, _ := psutilnet.Interfaces()
-		ifNames := ReadAppInterfaceList(aiStatus.DomainName)
+		ifNames := getAppInterfaceList(*aiStatus)
```
Comment (kalyan-nidumolu) [inline pkg/pillar/cmd/zedagent/handlemetrics.go]: same here..  aStatus. getAppInterfaceList()

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 1 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/zedagent/handlemetrics.go]: Fixed

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 2 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/zedagent/handlemetrics.go]: Fixed

File: pkg/pillar/cmd/zedagent/handlemetrics.go (hunk 3 above)
Comment (eriknordmark) [reply pkg/pillar/cmd/zedagent/handlemetrics.go]: Fixed

**Pull request merged**